### DISCUSSION TITLE
MS SQL Trusted Connection Datasource

### DISCUSSION THREAD

**Discussion by amrtdc1** [Budibase/budibase#3584] [Help] [answered]
First off...love what you all have done with Budibase...it looks very promising for some business use cases I have.  The only issue I have run into thus far is establishing a data connection.  I've tried and failed to establish a connection to an on-prem MS SQL server for which we utilize Trusted Connections (Windows credentials).  Looking into the code, it does not appear we currently have the option of using this type of authentication and would need to use SQL Server Authentication instead. Am I missing something?  Is it currently possible to utilize the Trusted Connection for authentication in the SQL Server Data Connection setup? If not, do you think this might be an option you can add as our permissions are heavily tied to the trusted connection (Windows login user)?

**Comment by mjashanks** [comment, marked as answer]
Thank you! Unfortunately we don't have the option for Trusted Connections. Though, it shouldn't be too hard to add, as the underlying libraries support it.

It would be a connection option when creating a new datasource.

I've logged this here: https://github.com/Budibase/budibase/issues/3601 

PS - the new MSSQL integration is about to be released, which will allow fetching of tables and eliminate a lot of the basic CRUD queries that are needed.

**Reply by amrtdc1** [reply]
Thanks Mike!  Very happy to hear about the potential to add the trusted connection option...that will be fantastic for our use case.  Also, great to hear about the MSSQL integration piece easing the query process.
